go/parquet/metadata/column_chunk.go

```diff
@@ -0,0 +1,385 @@
+func (c *ColumnChunkMetaData) FileOffset() int64 { return c.column.FileOffset }
+
+// FilePath gives the name of the parquet file if provided in the metadata
+func (c *ColumnChunkMetaData) FilePath() string { return c.column.GetFilePath() }
+
+// Type is the physical storage type used in the parquet file for this column chunk.
+func (c *ColumnChunkMetaData) Type() parquet.Type { return parquet.Type(c.columnMeta.Type) }
+
+// NumValues is the number of values stored in just this chunk
+func (c *ColumnChunkMetaData) NumValues() int64 { return c.columnMeta.NumValues }
+
+// PathInSchema is the full path to this column from the root of the schema including
+// any nested columns
+func (c *ColumnChunkMetaData) PathInSchema() parquet.ColumnPath {
+	return c.columnMeta.GetPathInSchema()
+}
+
+// Compression provides the type of compression used for this particular chunk.
+func (c *ColumnChunkMetaData) Compression() compress.Compression {
+	return compress.Compression(c.columnMeta.Codec)
+}
+
+// Encodings returns the list of different encodings used in this chunk
+func (c *ColumnChunkMetaData) Encodings() []parquet.Encoding { return c.encodings }
+
+// EncodingStats connects the order of encodings based on the list of pages and types
+func (c *ColumnChunkMetaData) EncodingStats() []PageEncodingStats {
+	ret := make([]PageEncodingStats, len(c.encodingStats))
+	for idx := range ret {
+		ret[idx].Encoding = parquet.Encoding(c.encodingStats[idx].Encoding)
+		ret[idx].PageType = c.encodingStats[idx].PageType
+	}
+	return ret
+}
+
+// HasDictionaryPage returns true if there is a dictionary page offset set in
+// this metadata.
+func (c *ColumnChunkMetaData) HasDictionaryPage() bool {
+	return c.columnMeta.IsSetDictionaryPageOffset()
+}
+
+// DictionaryPageOffset returns the location in the file where the dictionary page starts
+func (c *ColumnChunkMetaData) DictionaryPageOffset() int64 {
+	return c.columnMeta.GetDictionaryPageOffset()
+}
+
+// DataPageOffset returns the location in the file where the data pages begin for this column
+func (c *ColumnChunkMetaData) DataPageOffset() int64 { return c.columnMeta.GetDataPageOffset() }
+
+// HasIndexPage returns true if the offset for the index page is set in the metadata
+func (c *ColumnChunkMetaData) HasIndexPage() bool { return c.columnMeta.IsSetIndexPageOffset() }
+
+// IndexPageOffset is the location in the file where the index page starts.
+func (c *ColumnChunkMetaData) IndexPageOffset() int64 { return c.columnMeta.GetIndexPageOffset() }
+
+// TotalCompressedSize will be equal to TotalUncompressedSize if the data is not compressed.
+// Otherwise this will be the size of the actual data in the file.
+func (c *ColumnChunkMetaData) TotalCompressedSize() int64 {
+	return c.columnMeta.GetTotalCompressedSize()
+}
+
+// TotalUncompressedSize is the total size of the raw data after uncompressing the chunk
+func (c *ColumnChunkMetaData) TotalUncompressedSize() int64 {
```

Comment:
nit: should there be bloom filter info here as well?

Reply:
Ideally yes there should be bloom filter info, I just haven't implemented that yet. The "TotalUncompressedSize" is just the value reported in the metadata though, so if there is bloom filter info in the file, it would be counted too.

Reply:
added function to return `BloomFilterOffset`